orchagent/routeorch.cpp

```diff
@@ -1395,8 +1401,13 @@ bool RouteOrch::addRoute(RouteBulkContext& ctx, const NextHopGroupKey &nextHops)
                 }
                 else
                 {
-                    SWSS_LOG_INFO("Failed to get next hop %s for %s",
+                    SWSS_LOG_INFO("Failed to get next hop %s for %s, resolving neighbor",
                             nextHops.to_string().c_str(), ipPrefix.to_string().c_str());
+                    if (m_neighborToResolve.find(nexthop) == m_neighborToResolve.end()) // TODO: Allow retry for unresolved neighbors
```

Comment:
>> TODO: Allow retry for unresolved neighbors
This should be the responsibility of the nbrmgr until nbr entry is not created in the kernel, the possible reason could be that L3 interface could be admin/oper down, while we attempt to resolve nbr from nbrmgr.

Reply:
Each request for neighbor resolution should trigger one attempt of neighbor resolution. Ideally, the route orch needs a logic to trigger retry for neighbor resolution. However, there is no immediate need for this since the `arp_update` script in https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/arp_update#L56 would ping failed neighbor and allow the neighbor to get resolved once it is ready.

Reply:
sure, let's discuss this later.